### ISSUE TITLE
Upgrade to PyTorch 1.7 and Lightning 1.1.6

### ISSUE THREAD

**Issue by ant0nsc** [microsoft/InnerEye-DeepLearning#370]
* Pytorch 1.7 will allow to remove `np.count_nonzero` in `get_number_of_voxels_per_class`
* Lightning 1.1.6 will enable correctly logging the `SecondsPerBatch` metric on the training set (at the moment it is mysteriously swallowed, even though logged)

**Comment by ant0nsc** [comment]
Add more checks in `_test_model_train`: This should check that all metrics are both in train and test metrics, and both files have as many rows as there are epochs.

**Comment by ant0nsc** [comment]
Most work done in #411. Change to `torch.count_nonzero` in #417
